noxfile.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+
+import nox  # type: ignore
+import packaging.version
+
+
+def fetch_latest_tag(session: nox.Session) -> packaging.version.Version:
+    if "LATEST_TAG" in os.environ:
+        return packaging.version.Version(os.environ["LATEST_TAG"])
+
+    session.run("git", "fetch", "--tags", external=True)
+    out = session.run("git", "tag", "--list", external=True, silent=True)
+    assert out
+    return max(packaging.version.Version(tag) for tag in out.splitlines())
+
+
+def get_pre_migration(latest_tag: packaging.version.Version) -> str:
+    migrations_folder = (
+        pathlib.Path(__file__).parent / "procrastinate" / "sql" / "migrations"
+    )
+    migrations = sorted(migrations_folder.glob("*.sql"))
+    pre_migration: pathlib.Path | None = None
+    for migration in migrations:
+        mig_version = packaging.version.Version(migration.name.split("_")[0])
+        if mig_version > latest_tag and "_post_" in migration.name:
+            break
+
+        pre_migration = migration
+
+    assert pre_migration is not None
+    return pre_migration.name
+
+
+@nox.session
+def current_version_with_post_migration(session: nox.Session):
+    session.install("poetry")
+    session.run("poetry", "install", "--all-extras")
+    session.run("poetry", "run", "pytest")
+
+
+@nox.session
+def current_version_without_post_migration(session: nox.Session):
+    latest_tag = fetch_latest_tag(session)
+    pre_migration = get_pre_migration(latest_tag)
+
+    session.install("poetry")
+    session.run("poetry", "install", "--all-extras")
+    session.run(
+        "poetry",
+        "run",
+        "pytest",
+        f"--migrate-until={pre_migration}",
+        "./tests/acceptance",
+    )
+
+
+@nox.session
+def stable_version_without_post_migration(session: nox.Session):
```

Comment:
It's funny because in the end the noxfile solution ended up being not that far from my original solution. Of course, it's an excellent idea to use nox so that we don't have to bother with creating our venvs ourselves. But in the end, if you want to be sure to run exactly the tests you want with the procrastinate you want, you need a tmpdir, and run things inside.
- In my solution, we use the same pytest process for everything, and it calls subprocesses for procrastinate. Whereas in your solution, we call distinct pytest subprocesses for each of the 3 cases.
- In my solution, we don't need to install anything else than procrastinate in the venv, whereas in your solution, we install pytest in all venvs.
- In your solution, we keep existing acceptance tests. In my solution, we need to rewrite them.

I don't think any approach outshines the other one at 100%, and I'd be comfortable with either. We're just not going to end up with the same set of problems.